crates/oxc_linter/src/rules/react_hooks/exhaustive_deps.rs

```diff
@@ -0,0 +1,3454 @@
+use itertools::Itertools;
+use oxc_semantic::{ScopeId, Semantic, SymbolId};
+use oxc_span::GetSpan;
+use rustc_hash::FxHashSet;
+use std::hash::Hash;
+
+use oxc_diagnostics::OxcDiagnostic;
+
+use oxc_ast::{
+    ast::{
+        Argument, ArrayExpressionElement, ArrowFunctionExpression, BindingPatternKind,
+        CallExpression, ChainElement, Expression, Function, FunctionBody, IdentifierReference,
+        MemberExpression, StaticMemberExpression, VariableDeclarationKind,
+    },
+    match_expression,
+    visit::walk::walk_function_body,
+    AstKind, Visit,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{Atom, Span};
+use phf::phf_set;
+
+use crate::{
+    ast_util::{get_declaration_of_variable, get_enclosing_function},
+    context::LintContext,
+    rule::Rule,
+    AstNode,
+};
+
+fn missing_callback_diagnostic(hook_name: &str, span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!("React hook {hook_name} requires an effect callback."))
+        .with_label(span)
+        .with_help("Did you forget to pass a callback to the hook?")
+}
+
+fn dependency_array_required_diagnostic(hook_name: &str, span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!(
+        "React Hook {hook_name} does nothing when called with only one argument."
+    ))
+    .with_label(span)
+    .with_help("Did you forget to pass an array of dependencies?")
+}
+
+fn unknown_dependencies_diagnostic(hook_name: &str, span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!(
+      "React Hook {hook_name} received a function whose dependencies are unknown. Pass an inline function instead."
+    ))
+    .with_label(span)
```

Comment:
Should this be help text?
```suggestion
    OxcDiagnostic::warn(format!(
      "React Hook {hook_name} received a function whose dependencies are unknown."
    ))
    .with_help("Pass an inline function instead.")
    .with_label(span)
```